pkg/kcp/nuke/types/state.go

```diff
@@ -0,0 +1,50 @@
+package types
+
+import (
+	cloudcontrolv1beta1 "github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/common/actions/focal"
+	skrruntime "github.com/kyma-project/cloud-manager/pkg/skr/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ResourceKindState struct {
+	Kind    string
+	List    client.ObjectList
+	Objects []focal.CommonObject
+}
+
+type ProviderResourceKindState struct {
+	Kind     string
+	Provider cloudcontrolv1beta1.ProviderType
+	Objects  []ProviderResourceObject
+}
+
+type ProviderResourceObject interface {
+	GetId() string
+	GetObject() interface{}
+}
+
+type State struct {
```

Comment:
Can this be interface, and actual struct in parent nuke package like on other places? It's usual to extract general type definition as an interface and expose that for the users of the feature (here the `types` package), while internal implementation is kept as internal actual type (parent `nuke` package).